msu/hooks/skills/skill_container.nut

```diff
@@ -27,6 +27,74 @@
 		this.m.ScheduledChangesSkills.clear();
 	}
 
+	local add = o.add;
+	o.add = function( _skill, _order = 0 )
+	{
+		if (!_skill.isStacking())
+		{
+			local incrementAddedStack = function( _alreadyPresentSkill, _skillToAdd )
+			{
+				_alreadyPresentSkill.m.MSU_AddedStack += 1;
+				if (!::MSU.isNull(_skillToAdd.getItem()))
+				{
+					_alreadyPresentSkill.setItem(_skillToAdd.getItem());
+					foreach (i, itemSkill in _skillToAdd.getItem().m.SkillPtrs)
+					{
+						if (itemSkill.getID() == _skillToAdd.getID())
+						{
+							_skillToAdd.getItem().m.SkillPtrs[i] = _alreadyPresentSkill;
+							_skillToAdd.setItem(null);
+							break;
+						}
+					}
+				}
+
+				if (_alreadyPresentSkill.m.MSU_AddedStack > 0)
+				{
+					_alreadyPresentSkill.m.IsGarbage = false;
+				}
+			}
+
+			foreach (i, alreadyPresentSkill in this.m.Skills)
+			{
+				if (alreadyPresentSkill.getID() == _skill.getID())
+				{
+					incrementAddedStack(alreadyPresentSkill, _skill);
+					if (alreadyPresentSkill.m.MSU_AddedStack > 1) alreadyPresentSkill.onRefresh();
+					return;
+				}
+			}
+
+			foreach (i, alreadyPresentSkill in this.m.SkillsToAdd)
+			{
+				if (alreadyPresentSkill.getID() == _skill.getID())
+				{
+					incrementAddedStack(alreadyPresentSkill, _skill);
+					return;
+				}
+			}
+		}
+
+		add(_skill, _order);
+	}
+
+	local remove = o.remove;
+	o.remove = function( _skill )
+	{
+		if (_skill.m.MSU_AddedStack == 1) remove(_skill);
+		else _skill.removeSelf();
+	}
+
+	local removeByID = o.removeByID;
+	o.removeByID = function( _skillID )
+	{
+		local skill = this.getSkillByID(_skillID);
+		if (skill == null) return;
+
+		if (skill.m.MSU_AddedStack == 1) this.removeByID(_skillID);
```

Comment:
Bad recursion on accident, remove `this.`

Reply:
Nice catch